ledger/hash_stores/memory_hash_store.py

```diff
@@ -23,10 +23,10 @@ def readNode(self, pos):
         return self._nodes[pos - 1]
 
     def readLeafs(self, startpos, endpos):
-        return (n for n in self._leafs[startpos - 1:endpos - 1])
+        return [n for n in self._leafs[startpos - 1:endpos]]
```

Comment:
Won't the original code (generator) be more efficient for huge data?

Reply:
When I was trying to debug failing tests I noticed that database-backed implementation returns a list, so I returned list here as well so implementations match as close as possible. If some code tries to iterate twice or use indexes then it will fail with generators. As for resource efficiency - with this storage we already have all leaves in memory anyways.